### PR TITLE
fix: Check that code is not a precompile when checking code

### PR DESCRIPTION
**Pull request by roninjin10**
I noticed a footgun that if your precompile takes no arguments the entire function will fail to recognize the code exists. This is because it checks code.length and code.length returns the number of arguments in the case of a function. Fix by checking if the code is a function

### CODE REVIEW

**Comment by jochem-brouwer** [comment]
How did you end up finding this bug? Did you create a custom precompile?

**Comment by roninjin10** [comment]
Yea I found it while debugging why [this test wasn't failing](https://github.com/evmts/evmts-monorepo/pull/670/files#diff-bf6818a43572aeb70e72b3b8be250639c5bfc97b7b421c8007c0c60c9e52aeeaR57)

I'm building a app building framework called EVMts. It allows you to import solidity files directly into typescript files and then execute them with ethereumjs in the browser or node. [a user asked for the feature of adding precompiles](https://twitter.com/norswap/status/1724944291204665439) so he could use it to encode client functionality in an onchain contract is why I added the feature.

File: packages/evm/src/evm.ts [hunk 1]
```diff
@@ -251,7 +251,7 @@ export class EVM implements EVMInterface {
     // Load code
     await this._loadCode(message)
     let exit = false
-    if (!message.code || message.code.length === 0) {
+    if (!message.code || (typeof message.code !== 'function' && message.code.length === 0)) {
```
Comment (roninjin10) [inline packages/evm/src/evm.ts:254]: Happy to add a test if we want one

Comment (holgerd77) [reply packages/evm/src/evm.ts:254]: Really interesting find 🙏 and usage! Yes, it would be great if you can take the relevant parts of the test case over.

**Comment by holgerd77** [comment]
Just approved the test run so that we can see that this generally works.

**Comment by jochem-brouwer** [comment]
Have linted and added a test

File: packages/evm/src/evm.ts (hunk 1 above)
Comment (roninjin10) [reply packages/evm/src/evm.ts:254]: Apologies GitHub notifications are shambles I didn’t see this comment

**Review by acolytec3 — APPROVED** [review]
LGTM

**Comment by holgerd77** [comment]
Thanks for the continuation! 🙏🙂

**Comment by holgerd77** [comment]
This will be fixed in #3261

**Pull request merged**